lib/beacon/content.ex

```diff
@@ -1640,14 +1642,160 @@ defmodule Beacon.Content do
-      %Ecto.Changeset{data: %Component{}}
+      iex> change_component_attr(component_attr, %{name: "Header"})
+      %Ecto.Changeset{data: %ComponentAttr{}}
 
   """
-  @doc type: :components
+  @doc type: :component_attrs
   @spec change_component_attr(ComponentAttr.t(), map()) :: Changeset.t()
-  def change_component_attr(%ComponentAttr{} = component, attrs \\ %{}) do
-    ComponentAttr.changeset(component, attrs)
+  def change_component_attr(%ComponentAttr{} = component_attr, attrs \\ %{}) do
+    ComponentAttr.changeset(component_attr, attrs)
+  end
+
+  # COMPONENT SLOTS
+
+  @doc """
+  Returns an `%Ecto.Changeset{}` for tracking slot changes.
+
+  ## Example
+
+      iex> change_component_slot(component_slot, %{name: "Slot A"})
+      %Ecto.Changeset{data: %ComponentSlot{}}
+
+  """
+  @doc type: :component_slots
+  @spec change_component_slot(ComponentSlot.t(), map()) :: Changeset.t()
+  def change_component_slot(%ComponentSlot{} = slot, attrs \\ %{}) do
+    ComponentSlot.changeset(slot, attrs)
+  end
+
+  @doc """
+  Creates a new component slot and returns the component with updated `:slots` association.
+  """
+  @doc type: :component_slots
+  @spec create_slot_for_component(Component.t(), %{name: binary()}) ::
+          {:ok, Component.t()} | {:error, Changeset.t()}
+  def create_slot_for_component(component, attrs) do
+    changeset =
+      component
+      |> Ecto.build_assoc(:slots)
+      |> ComponentSlot.changeset(attrs)
+
+    with {:ok, %ComponentSlot{}} <- Repo.insert(changeset),
+         %Component{} = component <- Repo.preload(component, [slots: [:attrs]], force: true) do
+      {:ok, component}
+    end
+  end
+
+  @doc """
+  Updates a component slot and returns the component with updated `:slots` association.
+  """
+  @doc type: :component_slots
+  @spec update_slot_for_component(Component.t(), ComponentSlot.t(), map()) :: {:ok, Component.t()} | {:error, Changeset.t()}
+  def update_slot_for_component(component, slot, attrs) do
+    changeset = ComponentSlot.changeset(slot, attrs)
+
+    with {:ok, %ComponentSlot{}} <- Repo.update(changeset),
+         %Component{} = component <- Repo.preload(component, [slots: [:attrs]], force: true) do
+      {:ok, component}
+    end
+  end
+
+  @doc """
+  Deletes a component slot and returns the component with updated slots association.
+  """
+  @doc type: :component_slots
+  @spec delete_slot_from_component(Component.t(), ComponentSlot.t()) :: {:ok, Component.t()} | {:error, Changeset.t()}
+  def delete_slot_from_component(component, slot) do
+    with {:ok, %ComponentSlot{}} <- Repo.delete(slot),
+         %Component{} = component <- Repo.preload(component, [slots: [:attrs]], force: true) do
+      {:ok, component}
+    end
+  end
+
+  @doc """
+  Add a changeset error if the option value doesn't matches the type.
+  """
+  @spec validate_if_value_matches_type(Changeset.t(), String.t(), any(), atom()) :: Changeset.t()
```

Comment:
```suggestion
  @doc false
```

You can mark this function semi-private since it's not meant to be called by users, it's only internal.